addons/osio-addon/src/main/java/io/fabric8/launcher/osio/providers/DependencyParamConverter.java

```diff
@@ -0,0 +1,42 @@
+package io.fabric8.launcher.osio.providers;
+
+import org.apache.maven.model.Dependency;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ParamConverter;
+
+import static javax.json.Json.createArrayBuilder;
+import static javax.json.Json.createObjectBuilder;
+
+@ApplicationScoped
+public class DependencyParamConverter implements ParamConverter<Dependency> {
+
+    @Override
+    public Dependency fromString(final String dep) {
+        Dependency dependency = new Dependency();
+        try {
+            String[] groupIdArtifactIdVersion = dep.split(":");
```

Comment:
I really think you should validate if the input is valid instead of letting throw an ArrayIndexOutOfBoundsException here.